Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -148,7 +154,8 @@ def __init__(self,
                  isotropic = True,
                  split = False,
                  info = False, 
-                 strong_convexity_constant = 0):
+                 strong_convexity_constant = 0,
+                 warmstart=False):
```

Comment:
I'm wondering whether this should be the default. Probably it shouldn't now, unless we decide to break backwards compatibility.

Reply:
I agree not without further discussion as users will find that their results would change if they run on the default settings. If we change warmstart=True by default, the default max_iteration value will also need to change.

Reply:
I think leaving it as `False` could work as it matches the regularisation toolkit and doesn't break backward compatibility. But worrying about breaking backward compatibility with a noticeable improvement seems like a bad plan - we want the default option to provide the best experience for CIL.  I think it will be confusing to switch this after it's merged, so I would suggest we set it to `True` now, but consider not merging until there is a major release if the backward compatibility is an issue. 

This is obviously the best option for smaller datasets (1K cube would add 12GB to the peak use of PDHG) but is going to have a big impact on bigger datasets (2k cube would add ~100GB). 

I would imagine we're someway off using any of these algorithms for larger datasets and  our approach may be different for them (the memory focused recon class). So I would consider seeing the default to `True` now, with a statement that setting to `False` may help if memory is an issue for the user.

Reply:
100 Gb is quite impressive! And that's on top of the algorithm! It'll be impossible with PDHG, atm. Maybe we could consider storing `p2` as float16 rather than float32?

Reply:
Yeah, 3x the image data can add up fast.

I think it would be worth investigating how much we actually need to store here. But maybe that's a future enhancement. I think there's potential savings in other functions/algorithms along these lines with storing less information.